File: spec/python/test_index_to_param_eos.py
import unittest

from index_to_param_eos import IndexToParamEos

class TestIndexToParamEos(unittest.TestCase):
    def test_index_to_param_eos(self):
        r = IndexToParamEos.from_file("src/index_sizes.bin")

        self.assertEqual(r.qty, 3)

        self.assertEqual(r.sizes[0], 1)
        self.assertEqual(r.sizes[1], 8)
        self.assertEqual(r.sizes[2], 4)

        self.assertEqual(r.blocks[0].buf, "A")
        self.assertEqual(r.blocks[1].buf, "BBBBBBBB")
        self.assertEqual(r.blocks[2].buf, "CCCC")

File: spec/javascript/test_index_to_param_eos.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('IndexToParamEos', 'src/index_sizes.bin', function(r) {
  assert.equal(r.qty, 3);

  assert.equal(r.sizes[0], 1);
  assert.equal(r.sizes[1], 8);
  assert.equal(r.sizes[2], 4);

  assert.equal(r.blocks[0].buf, "A");
  assert.equal(r.blocks[1].buf, "BBBBBBBB");
  assert.equal(r.blocks[2].buf, "CCCC");
});
